src/container.js

```diff
@@ -49,12 +49,28 @@ export class InvocationHandler {
   */
   invoke(container: Container, dynamicDependencies?: any[]): any {
     return dynamicDependencies !== undefined
-      ? this.invoker.invokeWithDynamicDependencies(container, this.fn, this.dependencies, dynamicDependencies)
-      : this.invoker.invoke(container, this.fn, this.dependencies);
+      ? injectProperties(container, this.fn, this.invoker.invokeWithDynamicDependencies(container, this.fn, this.dependencies, dynamicDependencies))
+      : injectProperties(container, this.fn, this.invoker.invoke(container, this.fn, this.dependencies));
   }
 }
 
 /**
+* Injects property dependencies if the conventional `injectProperties` is defined.
+*/
+function injectProperties(container, fn, instance) {
+  if (fn.injectProperties !== undefined) {
+    let dependencies = fn.injectProperties;
+    for (let property in dependencies) {
+      instance[property] = container.get(dependencies[property]);
+    }
+  }
+  if (instance.postConstruct !== undefined) {
```

Comment:
At first I put it inside the first conditional block, but then the resulting *convention chain* (i.e. having a conventional function being called only in the case another convention applies) didn't convince me and on a second thought I imagined the scenario where a user might take advantage of the hook even without having properties injected, using it like an *initializer* method which could be called more than once after instantiation. But, for the sake of DI optimization, I think I could safely make it dependent on the presence of `injectProperties` again!